src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -2039,21 +2039,29 @@ protected override void BeginProcessing()
 #if UNIX
                         process.WaitForExit();
 #else
-                        _waithandle = new ManualResetEvent(false);
-
-                        // Create and start the job object
-                        ProcessCollection jobObject = new();
-                        if (jobObject.AssignProcessToJobObject(process))
+                        if (_credential is not null)
                         {
-                            // Wait for the job object to finish
-                            jobObject.WaitOne(_waithandle);
+                            // If we are running as a different user, we cannot use a job object, so just wait on the process
+                            process.WaitForExit();
```

Comment:
Is there any reason not to always use `process.WaitForExit()`?  Also if the user provides explicit credentials for the current user account, `process.WaitForExit()` will be used instead of the job object.  Is this Ok?

Reply:
`-Wait` will wait not just for the new process but any processes that the new process spawns. By waiting on the job object it will wait until all of them have finished rather than just the current one.

Reply:
I think the intent of the job object is to also wait on any child processes, but processes in for a different user can't be added to the job object so in that case we can only WaitForExit() of the initial process.  WG had discussed this and was ok with this difference.

Reply:
> but processes in for a different user can't be added to the job object

This is not true, did you read my last comment?